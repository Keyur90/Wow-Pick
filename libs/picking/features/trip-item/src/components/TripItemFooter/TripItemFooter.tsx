import React, { useState } from 'react';
import { Button as MuiButton, AppBar, Toolbar, Fab, makeStyles } from '@material-ui/core';
import { ArrowBackIos, ArrowForwardIos, Keyboard } from '@material-ui/icons';
import { BottomActions, Button, ActionDrawer } from '@rf2/ui';
import { ActionSheet } from './components/ActionSheet';
import { ToteItem } from '@rf2/picking/data/api-contracts';

interface TripItemFooterPropTypes {
  onItemPrevious: () => void;
  onItemNext: () => void;
  onWScanClick: () => void;
  isWScanEnabled: boolean;
  canNavigatePrevious: boolean;
  onToggleView: () => void;
  viewAll: boolean;
  toteItem: ToteItem;
  onResetCurrentSuppliedItem: () => void;
  onShowOrderInformation: () => void;
}

const useStyles = makeStyles((theme) => ({
  appBar: {
    top: 'auto',
    bottom: 0,
  },
  manualScan: {
    position: 'absolute',
    bottom: 136,
    right: theme.spacing(2),
  },
}));

const useToolbarStyles = makeStyles((theme) => ({
  regular: {
    background: '#303030',
    borderTop: `1px solid ${theme.palette.common.black}`,
    padding: 0,
    minHeight: 64,
    display: 'flex',
    alignItems: 'stretch',
    '& > button': {
      width: '33.333333%',
      padding: theme.spacing(0, 2),
      textAlign: 'center',
      borderRadius: 0,
      letterSpacing: 0,
      lineHeight: '1rem',
      '&:first-child': {
        borderRight: `1px solid ${theme.palette.common.black}`,
      },
      '&:last-child': {
        borderLeft: `1px solid ${theme.palette.common.black}`,
      },
    },
  },
}));

const useMoreOptionsButtonStyles = makeStyles(() => ({
  label: {
    textTransform: 'initial',
  },
}));

const useNavButtonStyles = makeStyles(() => ({
  iconSizeMedium: {
    '& > svg:first-child': {
      fontSize: '1rem',
    },
  },
  disabled: {
    '& > .MuiButton-label': {
      color: '#888',
    },
  },
}));

const TripItemFooter: React.FC<TripItemFooterPropTypes> = ({
  onItemPrevious,
  onItemNext,
  onWScanClick,
  isWScanEnabled,
  canNavigatePrevious,
  onToggleView,
  viewAll,
  toteItem,
  onResetCurrentSuppliedItem,
  onShowOrderInformation,
}) => {
  const classes = useStyles();
  const toolbarClasses = useToolbarStyles();
  const moreOptionsButtonClasses = useMoreOptionsButtonStyles();
  const useNavButtonClasses = useNavButtonStyles();
  const [isActionDrawerOpen, setIsActionDrawerOpen] = useState(false);

  const onToggleActionDrawer = (force?: boolean) => {
    setIsActionDrawerOpen(force ? force : !isActionDrawerOpen);
  };

  const onSubstitue = () => {
    // on substitue
  };

  const onFwdClick = () => {
    onItemNext();
  };

  return (
    <>
      <BottomActions isPrimary={false}>
        <Button variant={'outlined'} onClick={onSubstitue}>
          Substitute
        </Button>
      </BottomActions>
      <AppBar position="fixed" color="primary" className={classes.appBar}>
        <Toolbar classes={toolbarClasses}>
          <MuiButton
            classes={useNavButtonClasses}
            color="inherit"
            onClick={onItemPrevious}
            startIcon={<ArrowBackIos />}
            disabled={!canNavigatePrevious}
          >
            BACK
          </MuiButton>
          <MuiButton classes={moreOptionsButtonClasses} color="inherit" onClick={() => setIsActionDrawerOpen(true)}>
            More Options
          </MuiButton>
          <MuiButton classes={useNavButtonClasses} color="inherit" onClick={onFwdClick} endIcon={<ArrowForwardIos />}>
            FWD
          </MuiButton>
        </Toolbar>
      </AppBar>
      <ActionDrawer isOpen={isActionDrawerOpen} onToggle={onToggleActionDrawer}>
        <ActionSheet
          onClickViewUnSuppliedArticle={onToggleView}
          onToggleDrawer={onToggleActionDrawer}
          viewAll={viewAll}
          toteItem={toteItem}
          onResetCurrentSuppliedItem={onResetCurrentSuppliedItem}
          onShowOrderInformation={onShowOrderInformation}
        />
      </ActionDrawer>
      <div className={classes.manualScan} hidden={!isWScanEnabled}>
        <Fab color="secondary" aria-label="manual scan" onClick={onWScanClick}>
          <Keyboard />
        </Fab>
      </div>
    </>
  );
};

export { TripItemFooter };
