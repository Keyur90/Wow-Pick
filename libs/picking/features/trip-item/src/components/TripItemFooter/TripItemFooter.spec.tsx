import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { TripItemFooter } from './TripItemFooter';

describe('component TripItemFooter', () => {
  const toteItem = {
    id: '28391',
    toteId: '4',
    itemIndex: 0,
    orderNo: '140031681',
    articleId: '28391',
    aisle: '9',
    bay: '56',
    shelf: '2',
    orderedQuantity: 3,
    currentPage: 5,
    stockOnHand: 23,
    totalPages: 20,
    version: '0.0.1',
    price: 4.0,
    secondaryArticleId: '124577',
    personalShopperNotes:
      "Please only make a substitution if you really have too. As these are the best pumpkin cubes I've ever had.",
    suppliedDetails: [],
    article: {
      id: '123123',
      pricingUnit: 'EACH',
      brand: 'Brand',
      description: 'Description',
    },
  };

  test('renders TripItemFooter', () => {
    const { container } = render(
      <TripItemFooter
        onItemPrevious={() => null}
        onItemNext={() => null}
        onShowOrderInformation={() => null}
        canNavigatePrevious={false}
        onToggleView={() => null}
        viewAll={true}
        toteItem={toteItem}
        isWScanEnabled={true}
        onResetCurrentSuppliedItem={null}
        onWScanClick={null}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('has manual scan button', () => {
    const { getByRole } = render(
      <TripItemFooter
        onItemPrevious={() => null}
        onItemNext={() => null}
        onShowOrderInformation={() => null}
        canNavigatePrevious={false}
        onToggleView={() => null}
        viewAll={true}
        toteItem={toteItem}
        isWScanEnabled={true}
        onResetCurrentSuppliedItem={null}
        onWScanClick={null}
      />
    );

    expect(
      getByRole('button', {
        name: /manual scan/i,
      })
    ).toBeInTheDocument();
  });

  test('show more options', async () => {
    const { getByText, findByText } = render(
      <TripItemFooter
        onItemPrevious={() => null}
        onItemNext={() => null}
        onShowOrderInformation={() => null}
        canNavigatePrevious={false}
        onToggleView={() => null}
        viewAll={true}
        toteItem={toteItem}
        isWScanEnabled={true}
        onResetCurrentSuppliedItem={null}
        onWScanClick={null}
      />
    );

    fireEvent(
      getByText(/more options/i),
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      })
    );

    expect(await findByText(/exit trolley trip/i)).toBeInTheDocument();
    expect(await findByText(/change article location/i)).toBeInTheDocument();
    expect(await findByText(/reset current supplied/i)).toBeInTheDocument();
    expect(await findByText(/View unsupplied articles/i)).toBeInTheDocument();
  });
});
