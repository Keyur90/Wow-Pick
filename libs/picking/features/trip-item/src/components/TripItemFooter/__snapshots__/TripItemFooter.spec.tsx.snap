// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component TripItemFooter renders TripItemFooter 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-7 makeStyles-appBar-9 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiContainer-root makeStyles-root-10 MuiContainer-maxWidthLg"
    >
      <div
        class="makeStyles-content-8"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary makeStyles-outlinedPrimary-11 MuiButton-outlinedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Substitute
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-regular-3 MuiToolbar-gutters"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit Mui-disabled makeStyles-disabled-6 Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium makeStyles-iconSizeMedium-5"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
              />
            </svg>
          </span>
          BACK
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label makeStyles-label-4"
        >
          More Options
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          FWD
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium makeStyles-iconSizeMedium-5"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    class="makeStyles-manualScan-2"
  >
    <button
      aria-label="manual scan"
      class="MuiButtonBase-root MuiFab-root MuiFab-secondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 5H4c-1.1 0-1.99.9-1.99 2L2 17c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm-9 3h2v2h-2V8zm0 3h2v2h-2v-2zM8 8h2v2H8V8zm0 3h2v2H8v-2zm-1 2H5v-2h2v2zm0-3H5V8h2v2zm9 7H8v-2h8v2zm0-4h-2v-2h2v2zm0-3h-2V8h2v2zm3 3h-2v-2h2v2zm0-3h-2V8h2v2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
